include/finufft/finufft_eitherprec.h

```diff
@@ -38,6 +38,22 @@
 // the plan object pointed to... (doesn't need to be even defined here)
 #define FINUFFT_PLAN_S FINUFFTIFY(_plan_s)
 
+#if defined(FINUFFT_DLL) && (defined(_WIN32) || defined(__WIN32__))
+#  if defined(dll_EXPORTS)
+#    define FINUFFT_EXPORT __declspec(dllexport)
+#  else
+#    define FINUFFT_EXPORT __declspec(dllimport)
+#  endif
+#else
+#  define FINUFFT_EXPORT
+#endif
```

Comment:
Ho about having these lines as a fallback for make users?

Reply:
> Ho about having these lines as a fallback for make users?

What does `as a fallback for make users` mean? Incorporate both these lines and cmake  `GenerateExportHeader` solution?

Reply:
Yes, only in case of make these are included, otherwise we use the cmake way. If the plan is to phase out make eventually.

Reply:
> Yes, only in case of make these are included, otherwise we use the cmake way. If the plan is to phase out make eventually.

Incorporating both solutions seems complicate the code more, current macros solution works for both makefile and cmake, changing to `GenerateExportHeader` will only work for cmake 3.12 and beyond. The macro way seems to work fine with fftw. https://github.com/FFTW/fftw3/blob/d0ce926f1523d95daed48cd7c69572e068dbbfb3/api/fftw3.h#L71-L94

Are we  phasing out makefile eventually @ahbarnett ?